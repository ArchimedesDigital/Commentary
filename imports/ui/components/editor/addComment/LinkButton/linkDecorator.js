import { CompositeDecorator } from 'draft-js';

const Link = (props) => {
	const {url} = props.contentState.getEntity(props.entityKey).getData();
	const style = {
		color: '#3b5998',
		textDecoration: 'underline',
	};
	return (
		<a href={url} style={style}>
			{props.children}
		</a>
	);
};

function findLinkEntities(contentBlock, callback, contentState) {
	contentBlock.findEntityRanges(
		(character) => {
			const entityKey = character.getEntity();

			return (
				entityKey !== null &&
				contentState.getEntity(entityKey).getType() === 'LINK'
			);
		},
		callback
	);
}

const linkDecorator = new CompositeDecorator([{
	strategy: findLinkEntities,
	component: Link,
}]);


export default linkDecorator;
export { Link, findLinkEntities };
