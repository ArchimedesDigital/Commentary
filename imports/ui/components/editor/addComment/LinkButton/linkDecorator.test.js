import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import { Link } from './linkDecorator';

describe('linkDecarator', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<MuiThemeProvider>
					<Link />
				</MuiThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
