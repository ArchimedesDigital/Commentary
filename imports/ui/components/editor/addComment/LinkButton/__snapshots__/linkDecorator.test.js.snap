// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`linkDecarator renders correctly 1`] = `
<a
  href="http://test.url"
  style={
    Object {
      "color": "#3b5998",
      "textDecoration": "underline",
    }
  }
/>
`;
